docker-images/openemr/run_openemr.sh

```diff
@@ -1,9 +1,53 @@
 #!/bin/sh
+# to be run by root with OpenEMR root dir as the CWD
 
 # allows customization of openemr credentials, preventing the need for manual setup
 #  - Setting db parameters via MYSQL_HOST, MYSQL_USER, MYSQL_PASS
 #  - Setting first-time account creation via OE_USER, OE_PASS
 #  - All of these must be passed to automate the first-time setup process
+set -e
+
+auto_setup() {
+    chmod -R 600 .
+    php auto_configure.php -f \
+        iuser=$OE_USER \
+        iuserpass=$OE_PASS \
+        login=$MYSQL_USER \
+        rootpass=$MYSQL_ROOT_PASS \
+        pass=$MYSQL_PASS \
+        server=$MYSQL_HOST \
+
+    #do I need to activate mod_rewrite?
+    echo "OpenEMR configured. Setting user 'www' as owner of openemr/ and setting file/dir permissions to 400/500"
+    #set all directories to 500
+    find . -type d -print0 | xargs -0 chmod 500
+    #set all file access to 400
+    find . -type f -print0 | xargs -0 chmod 400
+
+    echo "Default file permissions and ownership set, allowing writing to specific directories"
+    chmod 700 run_openemr.sh
+    # Set file and directory permissions
+    #chmod 666 sites/default/sqlconf.php why?
+    chmod 600 interface/modules/zend_modules/config/application.config.php # should this actually be writable?
+    find sites/default/documents -type d -print0 | xargs -0 chmod 700
+    find sites/default/edi -type d -print0 | xargs -0 chmod 700
+    find sites/default/era -type d -print0 | xargs -0 chmod 700
+    find sites/default/letter_templates -type d -print0 | xargs -0 chmod 700
+    find interface/main/calendar/modules/PostCalendar/pntemplates/cache -type d -print0 | xargs -0 chmod 700
+    find interface/main/calendar/modules/PostCalendar/pntemplates/compiled -type d -print0 | xargs -0 chmod 700
+    find gacl/admin/templates_c -type d -print0 | xargs -0 chmod 700
```

Comment:
Just make sure this find/xarg stuff will apply to stuff that gets put in these places by openemr (for example, when openemr places a new document in sites/default/documents/1/<some file> ).
Also maybe a stupid question, but why 700 and not 600 for above stuff?

Reply:
I think i figured out that you did 700 instead of 600 since are directories :)

Reply:
Yeah it took me some time to figure out it needed to be 700 and not 600. I feel like I need to learn a bit more deeply why directories behave the way they do w.r.t. the execute bit.